src/printer.js

```diff
@@ -3837,19 +3828,38 @@ function printJSXElement(path, options, print) {
     return openingLines;
   }
 
+  // Convert `{" "}` to text nodes containing a space.
+  // This makes it easy to turn them into `jsxWhitespace` which
+  // can then print as either a space or `{" "}` when breaking.
+  n.children = n.children.map(child => {
```

Comment:
@vjeux I had a quick look at avoiding mutation, but it looks like I'd need to create a new path based off the existing path, and I'm not sure how to do that!

Any pointers?